Improve error handling on existing seed view

bgptr opened the pull request:
This diff improves the error handling on the existing seed view by changing the input's border color to red if the seed word is not at valid position.
![ksnip_20201001-200949](https://user-images.githubusercontent.com/52497040/94849056-21111200-0425-11eb-9126-3a948898fa4f.png)

vctt94 commented on app/components/views/GetStartedPage/CreateWalletPage/ExistingSeed/ExistingSeed.jsx:
```diff
@@ -53,7 +53,7 @@ const ExistingSeed = ({
 
   const onError = useCallback(
```
Can remove this use callback

vctt94 commented on app/components/views/GetStartedPage/CreateWalletPage/ExistingSeed/ExistingSeed.jsx:
```diff
@@ -70,21 +70,24 @@ const ExistingSeed = ({
-      if (seedErrorStr.includes(POSITION_ERROR)) {
-        seedErrorStr = fixPositionError(seedErrorStr, seedWord);
+      if (seedErrorStr.includes(POSITION_ERROR) && seedWord) {
+        const regexp = new RegExp(`${POSITION_ERROR} (\\d+)`, "g");
```
I think adding a comment here explaining what is happening would be good

vctt94 commented on app/components/views/GetStartedPage/CreateWalletPage/ExistingSeed/ExistingSeed.jsx:
```diff
@@ -70,21 +70,24 @@ const ExistingSeed = ({
+          };
+          setSeedWords(updatedSeedWords);
+        } else {
+          console.log("invalid position error msg format");
```
why this log?

vctt94 reviewed (commented):
Thanks for this PR. 

I left a couple of comments, but I like the direction this is going.

bgptr commented:
Thank you for your review. I fixed your comments.

vctt94 reviewed (commented):
Thanks for fixing my comments.

I have a couple more comments. 

I noticed it is possible when entering a word in its wrong place, the wallet throws an error, but it does not show right after it. It only shows after you entering the next word, example:

Here the absurd word already errored, as you can see at decrediton logs:

![image](https://user-images.githubusercontent.com/15069783/95206327-f45e5100-07bc-11eb-99b4-ea76e842a069.png)

But it only shows as wrong if I add the third word:

![image](https://user-images.githubusercontent.com/15069783/95206425-11931f80-07bd-11eb-8a0a-615a9abe2839.png)

And if I delete the word from its position, the field border gets missed: 

![image](https://user-images.githubusercontent.com/15069783/95206499-266fb300-07bd-11eb-8487-57864df5d90e.png)

 I also think it would be good keeping the error message instead of just making the word red, as it can be a bit confusing. Making it red and showing an error message like "word on position n is wrong" would make it clearer. What do you think?

bgptr commented:
Thank you for your review. I fixed the mentioned issues and cleaned up the component a bit. Also, I added the suggested error message. It is much more straightforward now.

vctt94 reviewed (approved):
LGTM. Good job on this PR!
